includes/payment-methods/class-wc-stripe-payment-request.php

```diff
@@ -906,6 +906,32 @@ public function should_show_payment_request_button() {
 			return false;
 		}
 
+		if ( $this->is_product() && $this->get_product()->get_type() === 'variable' ) {
+			$valid_stock_statuses = array_filter(
+				array_keys( wc_get_product_stock_status_options() ),
+				function ( $stock_status ) {
+					return 'outofstock' !== $stock_status;
+				}
+			);
+			$product_args         = [
+				'status'       => 'publish',
+				'parent'       => $this->get_product()->get_id(),
+				'type'         => 'variation',
+				'stock_status' => $valid_stock_statuses,
+				// All we need is for 1 variation to exist to determine
+				// whether product is "instock" or "onbackorder".
+				'limit'        => 1,
+			];
+
+			// Retrieve variable product's variations that are not 'outofstock' (e.g. 'instock', 'onbackorder').
+			$variations = wc_get_products( $product_args );
```

Comment:
If I read this correctly. Here we are trying to hide the button if **all** variations are out of stock right? We could leverage `$this->get_product()->get_available_variations()` like this:

```php
if ( $this->is_product() && $this->get_product()->get_type() === 'variable' ) {
  $stock_availability = array_column($this->get_product()->get_available_variations(), 'is_in_stock');
  return array_unique($stock_availability) !== [false];
}
```

Which will hide the PRB if all variations are out of stock.

Reply:
Pretty neat. Didn't know we already had `get_available_variations()`. Good call 👍